Adding an interface to return a list of a given debtor invoice

DedrickEnc opened the pull request:
In this PR :
I added a new route to receive request : `debtors/:uuid/invoices?balanced{0|1}`
I added a new function  called `fetchInvoices` to handle, the request
I added three test cases to test the API.

jniles commented on server/test/api/account.js:
```diff
@@ -57,7 +57,7 @@ describe('The account API, PATH : /accounts', function () {
     });
 
- it('METHOD : GET, PATH : /accounts?locked=0, It returns a list of unlocked accounts', function () {
+ it('METHOD : GET, PATH : /accounts?locked=0, It returns a listst of unlocked accounts', function () {
```
Typo should be fixed. :)

jniles commented on server/test/api/debtors.js:
```diff
@@ -53,4 +53,46 @@ describe('The /debtors API', function () {
+  });
+  
+  it('METHOD : GET, PATH : /debtors/:uuid/invoices?balanced=0 returns a list of unbalance invoices of a given debtor', function () {
+    return agent.get('/debtors/3be232f9-a4b9-4af6-984c-5d3f87d5c107/invoices?balanced=0')
```
Instead of hard-coding the UUID into this string, it would be more readable and maintainable to put the UUID in a variable and template it in later.  Something like this:

``` js
// at the top of the describe() block
var debtorUuid = '3be232f9-a4b9-4af6-984c-5d3f87d5c107';

// ... later in the test script ...
it ('some description', function () {
  var target = '/debtors/:uuid/invoices?balanced=0'.replace(':uuid', debtorUuid);
  return agent.get(target)
  .then(function (res) { /* test conditions, etc.. */ });
});
```

Alternative, if you don't like using `replace()` you can just concatenate `'/debtors/' + debtorUuid + '/invoices?balanced=0'`.

jniles commented on server/test/api/debtors.js:
```diff
@@ -53,4 +53,46 @@ describe('The /debtors API', function () {
       .catch(helpers.handler);
   });
+
+ it.skip('METHOD : GET, PATH : /debtors/:uuid/invoices?balanced=3 returns an error bad value sent', function () {
```
Why is this skipped?

DedrickEnc replied:
Because it is not important, I am removing it!

jniles commented on server/test/api/references.js:
```diff
@@ -101,7 +101,7 @@ describe('The reference API, PATH : /references', function () {
         return agent.get('/references/' + DELETABLE_REFERENCE_ID);
       })
-      .then(function (res) {
+      .then(function (res) {res
```
I'm pretty sure this will throw a syntax error.

jniles commented on server/models/test/data.sql:
```diff
@@ -268,3 +268,5 @@ INSERT INTO `sale` (`project_id`, `reference`, `uuid`, `cost`, `currency_id`, `d
 INSERT INTO `sale_item` (`sale_uuid`, `uuid`, `inventory_uuid`, `quantity`, `inventory_price`, `transaction_price`, `debit`, `credit`) VALUES
   ('957e4e79-a6bb-4b4d-a8f7-c42152b2c2f6','2e1332a7-3e63-411e-827d-42ad585ff518','cf05da13-b477-11e5-b297-023919d3d5b0',3,25.0000,25.0000,0.0000,75.0000);
 
+INSERT INTO `posting_journal` (uuid, project_id, fiscal_year_id, period_id, trans_id, trans_date, account_id, debit, credit, debit_equiv, credit_equiv, currency_id, deb_cred_uuid, deb_cred_type, inv_po_id, origin_id, user_id, pc_id) VALUES ("017dbe1e-c37c-11e5-a86e-843a4b0cdadc", 1, 1, 1, "HBB1", '2016-01-26', 3631, 75.0000, 0, 75.0000, 0, 2, "3be232f9-a4b9-4af6-984c-5d3f87d5c107", "D", "957e4e79 ...
```
Could you put this on multiple lines, split on each VALUE group inserted?

jniles commented on server/controllers/finance/debtors.js:
```diff
@@ -107,9 +172,13 @@ function isEmpty(array) {
-exports.invoices = function invoices(req, res, next) {
+
+/**
+ * exports.invoices = function invoices(req, res, next) {
```
Is this ever used again?  Should it be deleted instead of just commented?

jniles commented on server/config/routes.js:
```diff
@@ -369,7 +369,8 @@ exports.configure = function (app) {
   app.get('/debtors/groups/:uuid', debtors.groupDetails);
   // app.get('/debtors/:uuid', debtors.details);
-  app.get('/debtors/:uuid/invoices', debtors.invoices);
+  //app.get('/debtors/:uuid/invoices', debtors.invoices);
```
Could you delete this line instead of just commenting it out?

DedrickEnc committed:
Adding an interface to return a list of a given debtor invoice

In this PR :
I added a new route to receive request : ```debtors/:uuid/invoices?balanced{0|1}```
I added a new function  called fetchInvoices to handle, the request
I added three test cases to test the API.

DedrickEnc committed:
removing unnecessary instruction

DedrickEnc committed:
Amending the PR
